Editor/Scripts/MeshSyncServerInspector.cs

```diff
@@ -87,14 +87,20 @@ public void DrawServerSettings(MeshSyncServer t)
         }
     }
 
-    private void DrawInstanceSettings(MeshSyncServer t)
-    {
+    private void DrawInstanceSettings(MeshSyncServer t) {
         var style = EditorStyles.foldout;
-        style.fontStyle = FontStyle.Bold;
+        style.fontStyle        = FontStyle.Bold;
         t.foldInstanceSettings = EditorGUILayout.Foldout(t.foldInstanceSettings, "Instances", true, style);
-        if (t.foldInstanceSettings)
-        {
-            t.InstanceHandling = (BaseMeshSync.InstanceHandlingType)EditorGUILayout.EnumPopup("Instance handling", t.InstanceHandling);
+        if (t.foldInstanceSettings) {
+            var newInstanceHandling =
+                (BaseMeshSync.InstanceHandlingType)EditorGUILayout.EnumPopup("Instance handling", t.InstanceHandling);
+
+            if (t.InstanceHandling != newInstanceHandling &&
+                EditorUtility.DisplayDialog("Warning",
+                    "Changing the instance handling mode will delete any prefabs and previously synced objects for this server. Are you sure you want to do this?",
+                    "Yes", "No")) {
+                t.InstanceHandling = newInstanceHandling;
```

Comment:
Should there be an opt out option? it seems that there is a convenient overload that takes care of storing the user preference for the session or the machine, you just need to provide a key constant.

Reply:
Yeah, [EditorPrefs ](https://docs.unity3d.com/ScriptReference/EditorPrefs.html)or [SessionState](https://docs.unity3d.com/ScriptReference/SessionState.html).

In this case, I would say SessionState is more appropriate ?

Using ScriptableSingleton as shown in this [PR](https://github.com/unity3d-jp/MeshSync/pull/730) is also another alternative

Reply:
👆

Reply:
Done